biodatasets/msh_wsd/msh_wsd.py

```diff
@@ -0,0 +1,245 @@
+MEDLINE; totaling 37,888 ambiguity cases in 37,090 MEDLINE citations.
+
+Note from the Author how to load dataset:
+1) Download the file MSHCorpus.zip (Link "MSHWSD Data Set") from
+   https://lhncbc.nlm.nih.gov/ii/areas/WSD/collaboration.html
+2) Set kwarg data_dir to the path of the zip file
+"""
+
+
+import itertools as it
+import re
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import datasets
+
+from utils import schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@article{,
+    author={Jimeno-Yepes, Antonio J.
+    and McInnes, Bridget T.
+    and Aronson, Alan R.},
+    title={Exploiting MeSH indexing in MEDLINE to generate a data set for word sense disambiguation},
+    journal={BMC Bioinformatics},
+    year={2011},
+    month={Jun},
+    day={02},
+    volume={12},
+    number={1},
+    pages={223},
+    issn={1471-2105},
+    doi={10.1186/1471-2105-12-223},
+    url={https://doi.org/10.1186/1471-2105-12-223}
+}
+
+"""
+
+_DATASETNAME = "msh_wsd"
+
+_DESCRIPTION = """\
+Evaluation of Word Sense Disambiguation methods (WSD) in the biomedical domain is difficult because the available
+resources are either too small or too focused on specific types of entities (e.g. diseases or genes). We have
+developed a method that can be used to automatically develop a WSD test collection using the Unified Medical Language
+System (UMLS) Metathesaurus and the manual MeSH indexing of MEDLINE. The resulting dataset is called MSH WSD and
+consists of 106 ambiguous abbreviations, 88 ambiguous terms and 9 which are a combination of both, for a total of 203
+ambiguous words. Each instance containing the ambiguous word was assigned a CUI from the 2009AB version of the UMLS.
+For each ambiguous term/abbreviation, the data set contains a maximum of 100 instances per sense obtained from
+MEDLINE; totaling 37,888 ambiguity cases in 37,090 MEDLINE citations.
+"""
+
+_HOMEPAGE = "https://lhncbc.nlm.nih.gov/ii/areas/WSD/collaboration.html"
+
+_LICENSE = "DUA (UMLS)"
+
+_URLS = {_DATASETNAME: ""}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_DISAMBIGUATION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class MshWsdDataset(datasets.GeneratorBasedBuilder):
+    """Biomedical Word Sense Disambiguation (WSD)."""
+
+    uid = it.count(0)
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="msh_wsd_source",
+            version=SOURCE_VERSION,
+            description="MSH-WSD source schema",
+            schema="source",
+            subset_id="msh_wsd",
+        ),
+        BigBioConfig(
+            name="msh_wsd_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="MSH-WSD BigBio schema",
+            schema="bigbio_kb",
+            subset_id="msh_wsd",
+        ),
+    ]
+
```

Comment:
If I understood correctly from #364 you need to add this: 
```python
BUILDER_CONFIG_CLASS = BigBioConfig
```

Reply:
yes, for now I think we can use the solution from #364 with a subclass of `BigBioConfig` defined here with the defaults that you want and `BUILDER_CONFIG_CLASS` set to that class you define.